sklearn/naive_bayes.py

```diff
@@ -299,7 +319,20 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
         self : object
             Returns self.
         """
-        return self._partial_fit(X, y, classes, _refit=False,
+        # deprecation warning if classes passed:
+        if classes is not None:
+            warnings.warn("The classes argument will be removed in version"
+                          "0.21. A new classes argument has been added as a"
+                          "class parameter since version 0.19 which is"
+                          "recommended to be used now.", DeprecationWarning)
+
+        # Raise error if classes set in both initialization and in partial_fit
+        if (self.classes is not None) & (classes is not None):
+            raise ValueError("The classes argument was already set in"
+                             "initialization. Resetting it in call to"
+                             "partial_fit is not allowed as this argument"
+                             "will be deprecated in version 0.21")
+        return self._partial_fit(X, y, classes=classes, _refit=False,
```

Comment:
as we discussed earlier, if self.classes was defined, then user should not pass this argument. if classes was defined at __init__ and passed in partial fit, the line above will give an error.

if none is passed, then self.classes will be used. this is being done in "_check_partial_fit_first_call" which is called from _partial_fit.

why should tests fail?